backend/src/main/java/com/htecgroup/skynest/lambda/impl/UploadFileToExternalServiceLambda.java

```diff
@@ -1,16 +1,49 @@
 package com.htecgroup.skynest.lambda.impl;
 
+import com.dropbox.core.DbxException;
+import com.dropbox.core.InvalidAccessTokenException;
+import com.dropbox.core.v2.DbxClientV2;
+import com.dropbox.core.v2.files.Metadata;
+import com.htecgroup.skynest.event.UploadFileToExternalServiceEvent;
+import com.htecgroup.skynest.exception.lambda.DropboxFailedException;
+import com.htecgroup.skynest.exception.lambda.DropboxInvalidAccessTokenException;
 import com.htecgroup.skynest.lambda.Lambda;
 import com.htecgroup.skynest.lambda.LambdaType;
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
 import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
 
-// TODO: Instead of String, lambda accepts Event and uploads file to external service, yet to be
-// decided.
 
 @Component
-public class UploadFileToExternalServiceLambda implements Lambda<String, Boolean> {
+@Log4j2
+@AllArgsConstructor
+public class UploadFileToExternalServiceLambda
+    implements Lambda<UploadFileToExternalServiceEvent, Boolean> {
+
+  DbxClientV2 dropboxClient;
+
   @Override
-  public Boolean doLambdaFunction(String objects) {
+  public Boolean doLambdaFunction(UploadFileToExternalServiceEvent event) {
+    try {
+      MultipartFile file = event.getFileToUpload();
+      ByteArrayInputStream inputStream = new ByteArrayInputStream(file.getBytes());
+      String fileName = event.getFileToUpload().getOriginalFilename();
+      Metadata uploadMetaData =
+          dropboxClient
+              .files()
+              .uploadBuilder("/" + event.getBucketName() + "/" + fileName)
```

Comment:
should probably use `String.format()`

also, shouldn't this contain bucket ids, and not names? names aren't unique

Reply:
I didn't like it with ids, it's way more useless seeing that UUID there for the bucket..

Reply:
it is, but there could be two buckets with the same name
you could include both somehow

Reply:
The files just go in the same bucket on dropbox. We don't have path on dropbox anyhow. It's just the buckets.